sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1267,7 +1267,22 @@ colDefinitionOption
     ;
 
 generationExpression
-    : GENERATED ALWAYS AS LEFT_PAREN expression RIGHT_PAREN
+    : GENERATED ALWAYS AS LEFT_PAREN expression RIGHT_PAREN     #generatedColumn
+    | GENERATED (ALWAYS | BY DEFAULT) AS IDENTITY identityColSpec? #identityColumn
+    ;
+
+identityColSpec
+    : LEFT_PAREN identityColSpecOption* RIGHT_PAREN
+    ;
+
+identityColSpecOption
+    : START WITH start=identityStartOrStep
+    | INCREMENT BY step=identityStartOrStep
+    ;
+
+identityStartOrStep
+    : MINUS? INTEGER_VALUE
+    | MINUS? BIGINT_LITERAL
```

Comment:
W don't have a generic primitive for integral literals?

Reply:
It seems we don't. I cannot find it.